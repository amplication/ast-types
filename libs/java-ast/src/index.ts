// Core exports
export * from "./lib/core/AstNode";
export * from "./lib/core/Writer";

// AST nodes exports
export * from "./lib/ast/Access";
export * from "./lib/ast/Annotation";
export * from "./lib/ast/Class";
export * from "./lib/ast/ClassReference";
export * from "./lib/ast/CodeBlock";
export * from "./lib/ast/Enum";
export * from "./lib/ast/Field";
export * from "./lib/ast/GenericType";
export * from "./lib/ast/Import";
export * from "./lib/ast/Interface";
export * from "./lib/ast/Method";
export * from "./lib/ast/Parameter";
export * from "./lib/ast/Type";

// Utils exports
// (Add utility functions as needed)
