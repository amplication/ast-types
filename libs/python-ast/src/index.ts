export * from "./lib/ast";
export * from "./lib/core";
