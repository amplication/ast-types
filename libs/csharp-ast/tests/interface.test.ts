import { CsharpSupport as csharp } from "../src/";

describe("class", () => {
  it("basic", () => {
    const myInterface = csharp.interface_({
      name: "IMyInterface",
      namespace: "Amplication",
      partial: true,
      access: "public",
      isNestedInterface: true,
    });

    myInterface.addMethod(
      csharp.method({
        access: "public",
        name: "ToString()",
        isAsync: true,
        parameters: [
          csharp.parameter({
            name: "myParameter",
            type: csharp.Types.string(),
          }),
        ],
        return_: csharp.Types.string(),
      })
    );

    myInterface.addField(
      csharp.field({
        access: "protected",
        name: "myFieldDictionary",
        type: csharp.Types.map(csharp.Types.string(), csharp.Types.string()),
      })
    );

    expect(myInterface.name).toBe("IMyInterface");
    expect(myInterface.toString()).toMatchSnapshot();
  });

  it("with async and sync methods", () => {
    const myInterface = csharp.interface_({
      name: "IMyInterface",
      namespace: "Amplication",
      access: "public",
      isNestedInterface: true,
    });

    myInterface.addMethod(
      csharp.method({
        access: "public",
        name: "AsyncAction",
        isAsync: true,
        parameters: [
          csharp.parameter({
            name: "myParameter",
            type: csharp.Types.string(),
          }),
        ],
      })
    );

    myInterface.addMethod(
      csharp.method({
        access: "public",
        name: "SyncAction",
        isAsync: false,
        parameters: [
          csharp.parameter({
            name: "myParameter",
            type: csharp.Types.string(),
          }),
        ],
      })
    );

    myInterface.addMethod(
      csharp.method({
        access: "public",
        name: "AsyncStringAction",
        isAsync: true,
        parameters: [
          csharp.parameter({
            name: "myParameter",
            type: csharp.Types.string(),
          }),
        ],
        return_: csharp.Types.string(),
      })
    );

    myInterface.addMethod(
      csharp.method({
        access: "public",
        name: "SyncStringAction",
        isAsync: false,
        parameters: [
          csharp.parameter({
            name: "myParameter",
            type: csharp.Types.string(),
          }),
        ],
        return_: csharp.Types.string(),
      })
    );

    expect(myInterface.name).toBe("IMyInterface");
    expect(myInterface.toString()).toMatchSnapshot();
  });
});
