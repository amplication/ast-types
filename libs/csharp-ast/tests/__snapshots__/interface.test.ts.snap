// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class basic 1`] = `
"public partial interface IMyInterface
{
    protected Dictionary<string, string> myFieldDictionary;
    
    public Task<string> ToString()(string myParameter);
}
"
`;

exports[`class with async and sync methods 1`] = `
"public interface IMyInterface
{
    public Task AsyncAction(string myParameter);
    public void SyncAction(string myParameter);
    public Task<string> AsyncStringAction(string myParameter);
    public string SyncStringAction(string myParameter);
}
"
`;
